R/tmiic.plot.R

```diff
@@ -0,0 +1,444 @@
+#' # igraph::plot.igraph() is not optimal when the graph contains
+#' # multiple edges between the same nodes.
+#' # So the recommend way to plot a compact graph is to use tmiic:  
+#' flatten.res <- tmiic.flatten_network(tmiic.res)
+#' plot(flatten.res)
+#' }
+#'
+#' }
+#-----------------------------------------------------------------------------
+tmiic.export <- function (tmiic.res, method = "igraph") {
+  if (is.null(tmiic.res$all.edges.summary)) {
+    stop("The inferred network does not exist")
+  }
+  if (is.null(method)) {
+    stop("Plotting method is required")
+  }
+  if (method != "igraph") {
+    stop("Method not supported")
+  }
+  return(tmiic.getIgraph(tmiic.res))
+}
+
+#-----------------------------------------------------------------------------
+# tmiic.getIgraph
+#-----------------------------------------------------------------------------
+#' Igraph plotting function for tmiic (temporal mode of miic)
+#'
+#' @description This functions returns an igraph object built from the result
+#' returned by:
+#' \code{\link{miic}} executed in temporal temporal mode.
+#' \code{\link{tmiic.flatten_network}}
+#' 
+#' @details
+#' Edges attributes are passed to the igraph graph and can be accessed with
+#' e.g. \code{E(g)$partial_correlation}. See \code{\link{miic}} for more
+#' details on edge parameters. By default, edges are colored according to the
+#' partial correlation between two nodes conditioned on the conditioning set
+#' (negative is blue, null is gray and positive is red) and their width is
+#' based on the conditional mutual information minus the complexity cost.
+#'
+#' @param tmiic.res [a tmiic graph object]
+#' The graph object returned by the miic execution in temporal mode, 
+#' eventually flattened
+#'
+#' @return An igraph graph object.
+#' 
+#' @export
+#'
+#' @seealso
+#' \code{\link{miic}} for details on edge parameters in the returned object,
+#' \code{\link[igraph]{igraph.plotting}} for the detailed description of the
+#' plotting parameters and \code{\link[igraph]{layout}} for different layouts.
+#-----------------------------------------------------------------------------
+tmiic.getIgraph <- function (tmiic.res){
+  if (class(tmiic.res) != "tmiic") {
+    stop("Not a tmiic object.")
+  }
+  
+  class(tmiic.res) <- "miic"
+  graph <- getIgraph(tmiic.res)
+  class(tmiic.res) <- "tmiic"
+  
+  if ( tmiic.isLaggeg (colnames(tmiic.res$adj_matrix)) ) {
+    igraph::V(graph)$label.dist = 1
+    igraph::V(graph)$label.degree = pi/2
+    igraph::E(graph)$curved = TRUE
+  }
+  else {
+    igraph::E(graph)$curved = FALSE
+    igraph::E(graph)$label <- tmiic.res$all.edges.summary$lag
+  }
+  return(graph)
+}
+ 
+#-----------------------------------------------------------------------------
+# tmiic.isLaggeg
+#-----------------------------------------------------------------------------
+# Test if the list of nodes corresponds to a lagged or non lagged graph
+#
```

Comment:
In  temporal mode, we plot differently the lagged and flatten networks. The user does not need to care if what he asked to be plot is the lagged and flatten one as the function evaluate this form him.

Reply:
I don't quite get your point here. Isn't the mode of plot chosen by the user, why would they not care? If I understand your code correctly, this function is used only in `tmiic.getIgraph` and `plot.tmiic`, that is, only in temporal mode, and this should happen after the mode is chosen. So why not use the mode information instead? In other words, if your plot function has different modes, then be explicit and add the corresponding parameter.

Reply:
I added a "tmiic_specific" list in the object returned by miic with an "is_lagged" boolean. So the function will be dropped in the next push

Reply:
A list is good if you have plan for more parameters, but if you only want to specify that this parameter is for temporal mode, then there's no need, because it's already a `tmiic` object, so in that case you can add the parameter directly

Reply:
For now I have two items: "is_lagged" and "n_nodes_not_lagged".